Sklep/Utils/BarcodeScanner.cs

```diff
@@ -43,12 +43,13 @@ private void VideoCaptureDevice_NewFrame(object sender, AForge.Video.NewFrameEve
             Bitmap bitmap = (Bitmap)eventArgs.Frame.Clone();
             filter.ApplyInPlace(bitmap);
             var result = reader.Decode(bitmap);
-            if(pictureBox != null)
+            if (pictureBox != null)
             {
                 pictureBox.Image = bitmap;
             }
-            if(result != null && EANValidator.validateBarcode(result.ToString()) && (DateTime.Now.Subtract(lastScanTime).TotalSeconds > 2 || result.ToString()!=lastScannedBarcode))
+            if (result != null && EANValidator.validateBarcode(result.ToString()))
```

Comment:
no nie do końca o to mi chodziło, bo teraz rozbiłeś długiego ifa na dwa zagnieżdżone. `if(result==null || !EANValidator.validateBarcode(result.ToString()) return;` zajmie tyle samo miejsca, a nie stworzy kolejnego zagnieżdżenia